File: src/main/scala/laoflch/debezium/connector/informix/InformixTaskContext.scala
package laoflch.debezium.connector.informix

import io.debezium.connector.common.CdcSourceTaskContext
import io.debezium.schema.DataCollectionId

import java.util


/**
 * A state (context) associated with a Informix task
 *
 * @author laoflch Luo
 *
 */
class InformixTaskContext(val config: InformixConnectorConfig,
                          val schema: InformixDatabaseSchema)
  extends CdcSourceTaskContext(config.getContextName,
    config.getLogicalName,
    () => schema.tableIds.asInstanceOf[util.Collection[DataCollectionId]]) {
  // replace java schema::tableIds
}

File: src/main/java/laoflch/debezium/connector/informix/InformixTaskContext.java
/*
 * Copyright Debezium-Informix-Connector Authors.
 *
 * Licensed under the Apache Software License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
 */

package laoflch.debezium.connector.informix;

import io.debezium.connector.common.CdcSourceTaskContext;

public class InformixTaskContext extends CdcSourceTaskContext {
    public InformixTaskContext(InformixConnectorConfig config, InformixDatabaseSchema schema) {
        super(config.getContextName(), config.getLogicalName(), schema::tableIds);
    }
}
